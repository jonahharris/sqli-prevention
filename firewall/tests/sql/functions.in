>>> Sada testovacich SQL dotazu: funkce
# ADDDATE
SELECT ADDDATE('2018-12-24', 31);
SELECT d, ADDDATE(d, 10) FROM t1;
SELECT d, ADDDATE(d, INTERVAL 10 HOUR) FROM t1;
# Avg
SELECT AVG(a) FROM t1;
SELECT AVG(DISTINCT a) FROM t1;
# BIT
SELECT BIT_AND(a) FROM t1;
SELECT BIT_OR(a) FROM t1;
SELECT BIT_XOR(a) FROM t1;
# Char
SELECT CHAR(49);
SELECT CHAR('49');
SELECT CHAR(0x65 USING utf8);
SELECT CHAR(77,97,114,'105',97,'68',66);
SELECT CHAR(77,77.3,'77.3');
# Charset
SELECT CHARSET(CHAR(0x65)), CHARSET(CHAR(0x65 USING utf8));
# Concat
SELECT CONCAT('Mar', 'ia', 'DB');
SELECT CONCAT(surname,', ',name) AS full_name FROM authors ORDER BY full_name;
# Count
SELECT COUNT(*) FROM student;
SELECT COUNT(name) FROM student;
# CURDATE a CURTIME
SELECT CURDATE();
SELECT CURTIME();
# Date_add a date_sub
SELECT DATE_ADD('2008-12-31 23:59:59', INTERVAL 1 SECOND);
SELECT DATE_SUB('2005-01-01', INTERVAL 1 SECOND);
SELECT DATE_ADD('2000-12-31 23:59:59', INTERVAL 1 SECOND);
SELECT DATE_ADD('2010-12-31 23:59:59', INTERVAL 1 DAY);
SELECT DATE_ADD('2100-12-31 23:59:59', INTERVAL '1:1' MINUTE_SECOND);
SELECT DATE_ADD('1900-01-01 00:00:00', INTERVAL '-1 10' DAY_HOUR);
SELECT DATE_ADD('1992-12-31 23:59:59.000002', INTERVAL '1.999999' SECOND_MICROSECOND);
SELECT DATE_SUB('1998-01-02', INTERVAL 31 DAY);
SELECT DATE_SUB('2005-01-01 00:00:00', INTERVAL '1 1:1:1' DAY_SECOND);
# Max
SELECT MAX(a) FROM t1;
SELECT MAX(DISTINCT a) FROM t1;
# Mid
SELECT MID('abcd', 4, 1);
# Min
SELECT MIN(a) FROM t1;
SELECT MIN(DISTINCT a) FROM t1;
# Now
SELECT NOW();
# Subdate
SELECT SUBDATE('2018-12-24', 31);
SELECT d, SUBDATE(d, 10) FROM t1;
SELECT d, SUBDATE(d, INTERVAL 10 HOUR) FROM t1;
# Substring
SELECT SUBSTRING('Knowledgebase',5);
SELECT SUBSTRING('MariaDB' FROM 6);
SELECT SUBSTRING('Knowledgebase',3,7);
SELECT SUBSTRING('Knowledgebase', -4);
SELECT SUBSTRING('Knowledgebase', -8, 4);
SELECT SUBSTRING('Knowledgebase' FROM -8 FOR 4);
# Sum
SELECT SUM(a) FROM t1;
SELECT SUM(DISTINCT a) FROM t1;
# Trim
SELECT TRIM('  bar   ');
SELECT TRIM(LEADING 'x' FROM 'xxxbarxxx');
SELECT TRIM(BOTH 'x' FROM 'xxxbarxxx');
SELECT TRIM(TRAILING 'xyz' FROM 'barxxyz');
# Truncate
SELECT TRUNCATE(1.223,1);
SELECT TRUNCATE(1.999,1);
SELECT TRUNCATE(1.999,0);
SELECT TRUNCATE(-1.999,1);
SELECT TRUNCATE(122,-2);
# Ostatni
SELECT AVG(score), t1.* FROM t1;
SELECT user, MAX(salary) FROM users GROUP BY user HAVING MAX(salary) > 10;
