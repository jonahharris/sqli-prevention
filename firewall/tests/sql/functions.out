>>> Sada testovacich SQL dotazu: funkce
# ADDDATE
STRING '***';NUMBER 31;CALL 2 DATE_ADD;SELECTNODATA 0 1;STMT;1;0;
NAME d;NAME d;NUMBER 10;CALL 2 DATE_ADD;TABLE t1;SELECT 0 2 1;STMT;1;0;
NAME d;NAME d;NUMBER 10;INTERVAL HOUR;CALL 3 DATE_ADD;TABLE t1;SELECT 0 2 1;STMT;1;0;
# Avg
NAME a;CALL 1 AVG;TABLE t1;SELECT 0 1 1;STMT;1;0;
NAME a;CALL 2 AVG;TABLE t1;SELECT 0 1 1;STMT;1;0;
# BIT
NAME a; CALL 1 BIT_AND;TABLE t1;SELECT 0 1 1;STMT;1;0;
NAME a; CALL 1 BIT_OR;TABLE t1;SELECT 0 1 1;STMT;1;0;
NAME a; CALL 1 BIT_XOR;TABLE t1;SELECT 0 1 1;STMT;1;0;
# Char
NUMBER 49;CALL 1 CHAR;SELECTNODATA 0 1;STMT;1;0;
STRING '***';CALL 1 CHAR;SELECTNODATA 0 1;STMT;1;0;
STRING '***';CALL 1 CHAR with USING utf8;SELECTNODATA 0 1;STMT;1;0;
NUMBER 77;NUMBER 97;NUMBER 114;STRING '***';NUMBER 97;STRING '***';NUMBER 66;CALL 7 CHAR;SELECTNODATA 0 1;STMT;1;0;
NUMBER 77;FLOAT 77.3;STRING '***';CALL 3 CHAR;SELECTNODATA 0 1;STMT;1;0;
# Charset
STRING '***';CALL 1 CHAR; CALL 1 FCHARSET;STRING '***';CALL 1 CHAR with USING utf8; CALL 1 FCHARSET;SELECTNODATA 0 2;STMT;1;0;
# Concat
STRING '***';STRING '***';STRING '***';CALL 3 CONCAT;SELECTNODATA 0 1;STMT;1;0;
NAME surname;STRING '***';NAME name;CALL 3 CONCAT;ALIAS full_name;TABLE authors;NAME full_name;ORDERBY 0;ORDERBYLIST 1;SELECT 0 1 1;STMT;1;0;
# Count
COUNTALL;TABLE student;SELECT 0 1 1;STMT;1;0;
NAME name; CALL 1 COUNT;TABLE student;SELECT 0 1 1;STMT;1;0;
# CURDATE a CURTIME
CALL 0 CURDATE;SELECTNODATA 0 1;STMT;1;0;
CALL 0 CURTIME;SELECTNODATA 0 1;STMT;1;0;
# Date_add a date_sub
STRING '***';NUMBER 1;INTERVAL SECOND;CALL 3 DATE_ADD;SELECTNODATA 0 1;STMT;1;0;
STRING '***';NUMBER 1;INTERVAL SECOND;CALL 3 DATE_SUB;SELECTNODATA 0 1;STMT;1;0;
STRING '***';NUMBER 1;INTERVAL SECOND;CALL 3 DATE_ADD;SELECTNODATA 0 1;STMT;1;0;
STRING '***';NUMBER 1;INTERVAL DAY;CALL 3 DATE_ADD;SELECTNODATA 0 1;STMT;1;0;
STRING '***';STRING '***';INTERVAL MINUTE_SECOND;CALL 3 DATE_ADD;SELECTNODATA 0 1;STMT;1;0;
STRING '***';STRING '***';INTERVAL DAY_HOUR;CALL 3 DATE_ADD;SELECTNODATA 0 1;STMT;1;0;
STRING '***';STRING '***';INTERVAL SECOND_MICROSECOND;CALL 3 DATE_ADD;SELECTNODATA 0 1;STMT;1;0;
STRING '***';NUMBER 31;INTERVAL DAY;CALL 3 DATE_SUB;SELECTNODATA 0 1;STMT;1;0;
STRING '***';STRING '***';INTERVAL DAY_SECOND;CALL 3 DATE_SUB;SELECTNODATA 0 1;STMT;1;0;
# Max
NAME a;CALL 1 MAX;TABLE t1;SELECT 0 1 1;STMT;1;0;
NAME a;CALL 2 MAX;TABLE t1;SELECT 0 1 1;STMT;1;0;
# Mid
STRING '***';NUMBER 4;NUMBER 1;CALL 3 SUBSTR;SELECTNODATA 0 1;STMT;1;0;
# Min
NAME a;CALL 1 MIN;TABLE t1;SELECT 0 1 1;STMT;1;0;
NAME a;CALL 2 MIN;TABLE t1;SELECT 0 1 1;STMT;1;0;
# Now
CALL 0 NOW;SELECTNODATA 0 1;STMT;1;0;
# ADDDATE
STRING '***';NUMBER 31;CALL 2 DATE_SUB;SELECTNODATA 0 1;STMT;1;0;
NAME d;NAME d;NUMBER 10;CALL 2 DATE_SUB;TABLE t1;SELECT 0 2 1;STMT;1;0;
NAME d;NAME d;NUMBER 10;INTERVAL HOUR;CALL 3 DATE_SUB;TABLE t1;SELECT 0 2 1;STMT;1;0;
# Substring
STRING '***';NUMBER 5;CALL 2 SUBSTR;SELECTNODATA 0 1;STMT;1;0;
STRING '***';NUMBER 6;CALL 2 SUBSTR;SELECTNODATA 0 1;STMT;1;0;
STRING '***';NUMBER 3;NUMBER 7;CALL 3 SUBSTR;SELECTNODATA 0 1;STMT;1;0;
STRING '***';NUMBER -4;CALL 2 SUBSTR;SELECTNODATA 0 1;STMT;1;0;
STRING '***';NUMBER -8;NUMBER 4;CALL 3 SUBSTR;SELECTNODATA 0 1;STMT;1;0;
STRING '***';NUMBER -8;NUMBER 4;CALL 3 SUBSTR;SELECTNODATA 0 1;STMT;1;0;
# Sum
NAME a;CALL 1 SUM;TABLE t1;SELECT 0 1 1;STMT;1;0;
NAME a;CALL 2 SUM;TABLE t1;SELECT 0 1 1;STMT;1;0;
# Trim
STRING '***';CALL 1 TRIM;SELECTNODATA 0 1;STMT;1;0;
NUMBER 1;STRING '***';STRING '***';CALL 3 TRIM;SELECTNODATA 0 1;STMT;1;0;
NUMBER 3;STRING '***';STRING '***';CALL 3 TRIM;SELECTNODATA 0 1;STMT;1;0;
NUMBER 2;STRING '***';STRING '***';CALL 3 TRIM;SELECTNODATA 0 1;STMT;1;0;
# Truncate
FLOAT 1.223;CALL 1 TRUNCATE;SELECTNODATA 0 1;STMT;1;0;
FLOAT 1.999;CALL 1 TRUNCATE;SELECTNODATA 0 1;STMT;1;0;
FLOAT 1.999;CALL 0 TRUNCATE;SELECTNODATA 0 1;STMT;1;0;
FLOAT -1.999;CALL 1 TRUNCATE;SELECTNODATA 0 1;STMT;1;0;
NUMBER 122;CALL -2 TRUNCATE;SELECTNODATA 0 1;STMT;1;0;
# Ostatni
NAME score;CALL 1 AVG;FIELDNAME t1.*;TABLE t1;SELECT 0 2 1;STMT;1;0;
NAME user;NAME salary;CALL 1 MAX;TABLE users;NAME user;GROUPBY 0;GROUPBYLIST 1 0;NAME salary;CALL 1 MAX;NUMBER 10;CMP 2;HAVING;SELECT 0 2 1;STMT;1;0;
